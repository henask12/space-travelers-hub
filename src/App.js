import "./App.css";
import Navigation from "./components/Navigations/Navigation";

function App() {
    return (
        <div className="App">
            <Navigation />
        </div>
    );
}

export default App;
