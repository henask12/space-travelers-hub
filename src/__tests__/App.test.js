import { render } from '@testing-library/react';

// import { Provider } from 'react-redux'; // Import Provider from react-redux
import '@testing-library/jest-dom/extend-expect'; // For better DOM assertions
import { Provider } from 'react-redux';
import store from '../redux/store';
import App from '../App';

test('renders the App component correctly', () => {
  const { container } = render(<Provider store={store}><App /></Provider>);
  expect(container).toMatchSnapshot();
});
