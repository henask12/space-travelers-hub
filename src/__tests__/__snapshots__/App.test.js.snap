// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the App component correctly 1`] = `
<div>
  <div>
    <nav
      class="bg-white-800 py-4 border border-white-100 rounded-lg shadow-sm"
    >
      <div
        class="container mx-auto px-4 sm:px-8 md:px-16 lg:px-20 xl:px-24 2xl:px-32 flex justify-between items-center"
      >
        <a
          class="active flex items-center text-gray-600"
          href="#/"
        >
          <div
            class="nav__logo flex items-center mr-2"
          >
            <img
              alt="Space Travelers Logo"
              class="h-8 w-8"
              src="logo192.png"
            />
            <h1
              class="ml-2 font-bold text-lg"
            >
              Space Travelers Hub
            </h1>
          </div>
        </a>
        <div
          class="flex space-x-4"
        >
          <a
            class="text-gray-600 hover:text-black-900 font-bold"
            exact="true"
            href="/"
          >
            Missions
          </a>
          <a
            class="text-gray-600 hover:text-black-900 font-bold"
            exact="true"
            href="/rockets"
          >
            Rockets
          </a>
          <span>
            |
          </span>
          <a
            class="text-gray-600 hover:text-black-900 font-bold"
            exact="true"
            href="/profile"
          >
            My Profile
          </a>
        </div>
      </div>
    </nav>
    <div
      class="bg-white border border-gray-400 rounded-lg shadow-sm"
    >
      <div
        class="p-8"
      >
        <div
          class="bg-white border border-white-300 rounded-lg shadow-sm ml-8 mr-8"
        />
        <div
          class="flex flex-col"
        >
          <div
            class="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
          >
            <div
              class="py-2 align-middle inline-block min-w-full sm:px-6 lg:px-8"
            >
              <div
                class="shadow overflow-hidden border-b border-gray-200 sm:rounded-lg"
              >
                <table
                  class="m-6 min-w-full divide-y divide-gray-200"
                >
                  <thead
                    class="bg-gray-50"
                  >
                    <tr>
                      <th
                        class="px-6 py-3 text-left text-xs font-bold text-black-500 uppercase tracking-wider"
                        scope="col"
                      >
                        Name
                      </th>
                      <th
                        class="px-6 py-3 text-left text-xs font-bold text-black-500 uppercase tracking-wider"
                        scope="col"
                      >
                        Title
                      </th>
                      <th
                        class="px-6 py-3 text-left text-xs font-bold text-black-500 uppercase tracking-wider"
                        scope="col"
                      >
                        Status
                      </th>
                      <th
                        class="px-6 py-3 text-left text-xs font-bold text-black-500 uppercase tracking-wider"
                        scope="col"
                      >
                        Join
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="bg-white divide-y divide-gray-200"
                  />
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
